pyfar/signal.py

```diff
@@ -444,6 +193,416 @@ def __setitem__(self, key, value):
+        return self.times[-1]
+
+    @property
+    def times(self):
+        """Time instances the signal is sampled at."""
+        return self._times
+
+    def find_nearest_time(self, value):
+        """Returns the closest time index for a given time
+        Parameters
+        ----------
+        value : float, array-like
+            The times for which the indices are to be returned
+
+        Returns
+        -------
+        indices : int, array-like
+            The index for the given time instance. If the input was an array
+            like, a numpy array of indices is returned.
+        """
+        times = np.atleast_1d(value)
+        indices = np.zeros_like(times).astype(np.int)
+        for idx, time in enumerate(times):
+            indices[idx] = np.argmin(np.abs(self.times - time))
+        return np.squeeze(indices)
+
+
+class FrequencyData(Audio):
+    """Class for frequency data.
+
+    Objects of this class contain frequency data which is not directly
+    convertable to the time domain, i.e., non-equidistantly spaced bins or
+    incomplete spectra.
+
+    """
+    def __init__(self, data, frequencies, fft_norm=None, comment=None,
+                 dtype=np.complex):
+        """Init FrequencyData with data, and frequencies.
+
+        Attributes
+        ----------
+        data : ndarray, double
+            Raw data in the frequency domain. The memory layout of Data is 'C'.
+            E.g. data of shape (3, 2, 1024) has 3 x 2 channles with 1024
+            frequency bins each.
+        frequencies : ndarray, double
+            Frequencies of the data in Hz. The number of frequencies must match
+            the size of the last dimension of data.
+        fft_norm : 'none', 'unitary', 'amplitude', 'rms', 'power', 'psd'
+            The kind of Discrete Fourier Transform (DFT) normalization. See
+            pyfar.fft.normalization and _[1] for more information. The default
+            is 'none', which is typically used for energy signals, such as
+            impulse responses.
+        comment : str
+            A comment related to the data. The default is None.
+        dtype : string, optional
+            Raw data type of the signal. The default is float64
+
+        References
+        ----------
+        .. [1] J. Ahrens, C. Andersson, P. Höstmad, and W. Kropp, “Tutorial on
+               Scaling of the Discrete Fourier Transform and the Implied
+               Physical Units of the Spectra of Time-Discrete Signals,” Vienna,
+               Austria, May 2020, p. e-Brief 600.
+
+        """
+
+        Audio.__init__(self, 'freq', comment, dtype)
+
+        # init data
+        self._data = np.atleast_2d(np.asarray(data, dtype=np.complex))
+
+        # init frequencies
+        self._frequencies = np.atleast_1d(np.asarray(frequencies).flatten())
+        if self._frequencies.size != self.n_bins:
+            raise ValueError(
+                "The number of freqencies must be data.shape[-1]")
```

Comment:
The monotonous increase check from initializing times in TimeData would also make sense for frequencies

Reply:
Good point. Added this to the init and made an additional test.